packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts

```diff
@@ -87,6 +87,46 @@ export class AlbControllerVersion {
    */
   public static readonly V2_4_1 = new AlbControllerVersion('v2.4.1', false);
 
+  /**
+   * v2.4.2
+   */
+  public static readonly V2_4_2 = new AlbControllerVersion('v2.4.2', false);
+
+  /**
+   * v2.4.3
+   */
+  public static readonly V2_4_3 = new AlbControllerVersion('v2.4.3', false);
+
+  /**
+   * v2.4.4
+   */
+  public static readonly V2_4_4 = new AlbControllerVersion('v2.4.4', false);
+
+  /**
+   * v2.4.5
+   */
+  public static readonly V2_4_5 = new AlbControllerVersion('v2.4.5', false);
+
+  /**
+   * v2.4.6
+   */
+  public static readonly V2_4_6 = new AlbControllerVersion('v2.4.6', false);
+
+  /**
+   * v2.4.7
+   */
+  public static readonly V2_4_7 = new AlbControllerVersion('v2.4.7', false);
+
+  /**
+   * v2.5.0
+   */
+  public static readonly V2_5_0 = new AlbControllerVersion('v2.5.0', false);
+
+  /**
+   * v2.5.1
+   */
+  public static readonly V2_5_1 = new AlbControllerVersion('v2.5.1', false);
```

Comment:
I have tried using `.of('v2.5.1')` with a custom policy. It is able to deploy, however, the pods cannot start up because they don't have the required permissions for the K8s Lease API that were added in version 1.5.0 of the [helm chart](https://github.com/aws/eks-charts/commit/0541e76e0d425c01141c016eca90338bb6d348fa). I would suggest adding an extra constructor parameter for the chart version. The appropriate chart can then be picked per ALB version instead of the hardcoded `1.4.1`.

Reply:
Thanks for catching this! I will work on addressing it.

Reply:
Because of this, it's not good enough to default helm version to 1.4.1 everywhere. That would mean that if someone wanted to do this:

```ts
new AlbController(this, 'blah', {
  version: AlbControllerVersion.V2_5_1,
});
```

Somewhere along the line it would fail because it's not compatible with helm 1.4.1. So we'd force the user to have to do something like:

```ts
new AlbController(this, 'blah', {
  version: AlbControllerVersion.V2_5_1,
  helmChart: HelmChartVersion.V1_5_0,
});
```

Even though we know what helm chart version would be compatible. So we should be able to add the default in for the user.

In practice that means that we add a prop to the constructor of AlbControllerVersion called `helmChartVersion`.

```ts
class AlbControllerVersion {
  public static readonly V2_5_1 = new AlbControllerVersion('v2.5.1', HelmChartVersion.V1.5.0, false);
  private constructor(public readonly version: string, public readonly helmChart, public readonly custom: boolean) {}
}

// where we need it, we can do
props.albControllerVersion.helmChart;
```

None of this should be a breaking change since the constructor for AlbContrllerVersion is private

Reply:
Thanks, Kaizen. This makes sense. I've what I think are the necessary updates for the AlbControllerVersion class to support the addition of helmChartVersion in the constructor